iroh-net/src/magic_endpoint.rs

```diff
@@ -18,28 +18,92 @@ use crate::{
 
 pub use super::magicsock::EndpointInfo as ConnectionInfo;
 
-/// Address information for a node.
+/// A peer and it's addressing information.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct NodeAddr {
+pub struct PeerData {
```

Comment:
Why not `PeerAddr`? To me having `addr`  in there suggests "this is what you need to connect", which is what it is.
So if we decided for `Peer`, not `Node`, then I'd say let's do `PeerAddr`.
Returning `PeerAddr` from `endpoint.my_addr()` sounds good too.